### ISSUE TITLE
Talk can not be set as default app on Nextcloud 28 using the app order

### ISSUE THREAD

**Issue by susnux** [nextcloud/spreed#11134]
With Nextcloud 28 you can set a custom app order on a per-user basis, this also works with Talk.
But using that feature it is currently not possible to set Talk as the default app if sorted first, because Talk uses a closure to register its navigation entry instead of an appinfo entry.

To fix this talk needs to also provide the app key in the navigation entry (`"app" => "spreed"`):

https://github.com/nextcloud/spreed/blob/0cc7077bca7800dc9b7434b59a4c60f292673774/lib/AppInfo/Application.php#L358

**Comment by nickvergessen** [comment]
What's the difference between app and id?
Couldn't the code that fails fall back to id when app is missing?

**Comment by susnux** [comment]
>  Does the code that manages the default app fall back to another app then?

Yes the app needs to be enabled for that user and in the closure case it must resolve.

**Comment by susnux** [comment]
I was pretty sure we check for `type === 'link'`

**Comment by susnux** [comment]
> Well it's an API break/missing feature which could be solved in 99% of the cases with a fallback :) That's why I'm asking.

Well we could also check if ID is a valid app and use that as a fallback, seems a bit hacky but might work 99%.

**Comment by nickvergessen** [comment]
> I was pretty sure we check for `type === 'link'`

Well not in here:
https://github.com/nextcloud/server/blob/aa5f037af71c915424c6dcfd5ad2dc82797dc0d6/lib/private/App/AppManager.php#L834-L849
And that is what is used in the redirect respoonse

**Comment by joshtrichards** [comment]
Related: nextcloud/server#52420
